proposals/records-wip.md

```diff
@@ -76,3 +76,45 @@ each primary constructor parameter to the corresponding property of the other ty
 ```C#
 override Equals(object o) => Equals(o as T);
 ```
+
+## `with` expression
+
+A `with` expression is a new expression using the following syntax.
+
+```antlr
+with_expression
+    : switch_expression
+    | switch_expression 'with' anonymous_object_initializer
+```
+
+A `with` expression allows for "non-destructive mutation", designed to
+produce a copy of the receiver expression with modifications to properties
+listed in the `anonymous_object_initializer`.
+
+A valid `with` expression has a receiver with a non-target-typed, non-null type. The receiver type must contain an instance method called `With` with
```

Comment:
>non-target-typed, non-null type [](start = 48, length = 31)

These are not concepts in the language.  If you don't define a target for the conversion, then any expression in the structure of a program isn't target-typed.